csp/decorators.py

```diff
@@ -1,54 +1,89 @@
 from functools import wraps
 
 
-def csp_exempt(f):
-    @wraps(f)
-    def _wrapped(*a, **kw):
-        r = f(*a, **kw)
-        r._csp_exempt = True
-        return r
+def csp_exempt(REPORT_ONLY=None):
+    if callable(REPORT_ONLY):
+        raise RuntimeError(
+            "Incompatible `csp_exempt` decorator usage. This decorator now requires arguments, "
+            "even if none are passed. Change bare decorator usage (@csp_exempt) to parameterized "
+            "decorator usage (@csp_exempt()). See the django-csp 4.0 migration guide for more "
+            "information."
+        )
 
-    return _wrapped
+    def decorator(f):
+        @wraps(f)
+        def _wrapped(*a, **kw):
+            r = f(*a, **kw)
```

Comment:
Micro-nit, non-blocking: I personally prefer expressive variable names, so maybe `response` instead of `r`?